File: coffee-time-challenges/01-two-bases/main.py
#!/usr/bin/env python3

from itertools import combinations_with_replacement


def check(x, y, z):
    return x*100+y*10+z == x+y*9+z*9**2


def main():
    retults = []
    for x, y, z in combinations_with_replacement(list(range(10)), 3):
        if check(x, y, z):
            retults.append((x, y, z))
    return retults

if __name__ == '__main__':
    print(main())

File: coffee-time-challenges/01-two-bases/main.rs
fn main () {
    for x in range(0i, 9) {
        for y in range(0i, 9) {
            for z in range(0i, 9) {
                if check(x, y, z) {
                    println!("{} {} {}", x, y, z);
                }
            }
        }
    }
}

fn check (x: int, y: int, z: int) -> bool {
    x * 10 * 10 + y * 10 + z == z * 9 * 9 + y * 9 + x
}
